.github/workflows/basics.yml

```diff
@@ -34,7 +34,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: cargo doc
-        run: cargo doc --no-deps
+        run: /bin/bash -c "cargo doc 2>&1 > /dev/null | grep -q "warning:" ; test $? -ne 0"
```

Comment:
https://stackoverflow.com/questions/73981385/make-cargo-doc-fail-if-there-are-warnings

Could you try `RUSTDOCFLAGS` as suggested here?

Reply:
yeah, I did see that. It didn't seem to work. `cargo doc` still returned exit code 0 with warnings. Maybe I need to try more. I will check back again tomorrow

Reply:
It works now. I just mustn't have set the environment variable correctly. Resolving this conversation.